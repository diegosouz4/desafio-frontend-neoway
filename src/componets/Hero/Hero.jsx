import Breadcrumb from "../Breadcrumb/Breadcrumb";
import PropTypes from "prop-types";
import style from "./Hero.module.scss";
import PostInfo from "../PostInfo/PostInfo";

export default function Hero({
  author,
  publishedAt,
  title,
  urlToImage,
  likes,
}) {
  return (
    <section
      className={style.hero}
      style={{ backgroundImage: `url(${urlToImage})` }}
      aria-label="hero section"
    >
      <div className={`container ${style.container}`}>
        <div>
          <h1>{title}</h1>
          <Breadcrumb className={style.breadCrumb} title={title} />
          <PostInfo
            author={author}
            publishedAt={publishedAt}
            likes={likes}
            className={style.info}
          />
        </div>
      </div>
    </section>
  );
}

Hero.propTypes = {
  title: PropTypes.string.isRequired,
  author: PropTypes.string,
  publishedAt: PropTypes.string,
  urlToImage: PropTypes.string,
  likes: PropTypes.bool,
};
