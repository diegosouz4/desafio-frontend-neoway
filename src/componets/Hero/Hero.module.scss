.hero {
  background: var(--card-thumb) no-repeat center center;
  background-size: cover;
  border-radius: 0.25rem;
  box-shadow: 0 0.625rem 1.25rem -0.25rem rgba(0, 0, 0, 0.25);
  height: 350px;
  max-width: 1000px;
  margin: 1.25rem auto;
  overflow: hidden;
  width: 90%;

  position: relative;
  &::before {
    content: "";
    display: block;
    width: 100%;
    height: 100%;
    background-color: rgba(54, 12, 66, 0.7);
    backdrop-filter: blur(2.5px);

    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
  }

  .container {
    text-align: center;
    height: 100%;
    width: 90%;
    max-width: 600px;

    display: flex;
    align-items: center;
    justify-content: center;

    position: relative;
    z-index: 1;

    h1 {
      font-size: 1.125rem;
      line-height: 1.2;
      margin-bottom: 0;
      @media (min-width: 768px) {
        font-size: 2rem;
      }
    }

    .breadCrumb {
      margin-top: 1rem;
    }

    .info {
      width: 90%;
      max-width: 300px;
      margin-top: 1.25rem;
      margin-left: auto;
      margin-right: auto;

      display: flex;
      flex-wrap: wrap;
      align-items: center;
      justify-content: center;
      gap: 1rem;
      span {
        color: var(--color-white-100);
        display: inline-block;
        font-size: 0.75rem;
        font-weight: 400;
        opacity: 0.8;

        svg {
          display: inline-block;
          height: auto;
          margin-right: 4px;
          width: 0.875rem;

          position: relative;
          transform: translateY(3px);
        }
      }
    }
  }
}
