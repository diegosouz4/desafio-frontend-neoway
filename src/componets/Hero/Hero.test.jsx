import { render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Hero from "./Hero";

const mockInfo = {
  title: "Jest test title",
  author: "Jest author",
  publishedAt: "2023-10-05",
  urlToImage: "sample-image.jpg",
  likes: false,
};

describe("<Hero />", () => {
  it("hero", () => {
    const { asFragment } = render(
      <BrowserRouter>
        <Hero {...mockInfo} />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
