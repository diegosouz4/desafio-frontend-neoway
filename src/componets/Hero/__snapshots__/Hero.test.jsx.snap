// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> hero 1`] = `
<DocumentFragment>
  <section
    aria-label="hero section"
    style="background-image: url(sample-image.jpg);"
  >
    <div
      class="container undefined"
    >
      <div>
        <h1>
          Jest test title
        </h1>
        <div
          class="undefined undefined"
        >
          <a
            href="/"
          >
            home
          </a>
          <span>
            Jest test title
          </span>
        </div>
        <div>
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
              />
            </svg>
             2023/10/05
          </span>
          <span>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256A128 128 0 1 0 224 0a128 128 0 1 0 0 256zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
              />
            </svg>
             Jest author
          </span>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
